src/API/WesternStatesWater.WestDaat.Engines/ValidationEngine.cs

```diff
@@ -525,6 +526,38 @@ private ErrorBase ValidateApplicationDocumentUploadSasTokenRequest(ApplicationDo
         return null;
     }
 
+    private async Task<ErrorBase> ValidateApplicationDocumentDownloadSasTokenRequest(ApplicationDocumentDownloadSasTokenRequest request, ContextBase context)
+    {
+        var userContext = _contextUtility.GetRequiredContext<UserContext>();
+
+        var documentExistsRequest = new DTO.ApplicationSupportingDocumentExistsRequest
+        {
+            WaterConservationApplicationDocumentId = request.WaterConservationApplicationDocumentId
+        };
+
+        var documentExistsResponse = (DTO.ApplicationSupportingDocumentExistsResponse)await _applicationAccessor.Load(documentExistsRequest);
+
+        if (!documentExistsResponse.DocumentExists)
+        {
+            return CreateNotFoundError(context, $"WaterConservationApplicationDocument with Id ${request.WaterConservationApplicationDocumentId}");
+        }
+
+        var permissions = _securityUtility.Get(new DTO.PermissionsGetRequest { Context = context });
+        var orgPermissions = _securityUtility.Get(new DTO.OrganizationPermissionsGetRequest
+        {
+            Context = context,
+            OrganizationId = documentExistsResponse.FundingOrganizationId
+        });
+
+        if (userContext.UserId != documentExistsResponse.ApplicantId && !permissions.Contains(Permissions.ApplicationReview) &&
+            !orgPermissions.Contains(Permissions.ApplicationReview))
+        {
+            return CreateForbiddenError(new ApplicationDocumentDownloadSasTokenRequest(), context);
+        }
```

Comment:
Maybe split the logic check and early return if we can so it's more clear. Also, I'm not sure we need to check for non-org permissions as the org permissions should be enough to cover the global admin use case

```suggestion csharp
// Current user owns the application, so allow file download
if (userContext.UserId == documentExistsResponse.ApplicantId)
{
    return null;
}

var orgPermissions = _securityUtility.Get(new DTO.OrganizationPermissionsGetRequest
{
    Context = context,
    OrganizationId = documentExistsResponse.FundingOrganizationId
});

// Check if current user is a member of the application funding organization
if (!orgPermissions.Contains(Permissions.ApplicationReview))
{
    return CreateForbiddenError(new ApplicationDocumentDownloadSasTokenRequest(), context);
}
```